weld/llvm.rs

```diff
@@ -3386,8 +3444,8 @@ fn llvm_lt(k: ScalarKind) -> &'static str {
 }
 
 /// Returns an LLVM formatted String for a literal.
-fn llvm_literal(k: LiteralKind) -> String {
-    match k {
+fn llvm_literal(k: LiteralKind) -> WeldResult<String> {
```

Comment:
Might make more sense to move this into `LlvmGenerator` as a function now, and then call `gen_string_definition` etc. when matching on `StringLiteral`?

Reply:
I think I'm actually going to keep these separate, because e.g. `string_literal` requires passing in function state while `llvm_literal` doesn't.